float32 x_dist
float32 y_dist
float32 z_dist
float32 r
float32 p
float32 y
bool isClosed
